<h1>Summary</h1>
<p>A program can use exceptions to indicate that an error occurred. To throw an exception, use the <code>throw</code> statement and provide it with an exception object &#151; a descendant of <code>Throwable</code> &#151; to provide information about the specific error that occurred. A method that throws an uncaught, checked exception must include a <code>throws</code> clause in its declaration.</p>
<p>A program can catch exceptions by using a combination of the <code>try</code>, <code>catch</code>, and <code>finally</code> blocks.</p>
<ul>
<li>The <code>try</code> block identifies a block of code in which an exception can occur.</li>
<li>The <code>catch</code> block identifies a block of code, known as an exception handler, that can handle a particular type of exception.</li>
<li>The <code>finally</code> block identifies a block of code that is guaranteed to execute, and is the right place to close files, recover resources, and otherwise clean up after the code enclosed in the <code>try</code> block.</li>
</ul>
<p>The <code>try</code> statement should contain at least one <code>catch</code> block or a <code>finally</code> block and may have multiple <code>catch</code> blocks.</p>
<p>The class of the exception object indicates the type of exception thrown. The exception object can contain further information about the error, including an error message. With exception chaining, an exception can point to the exception that caused it, which can in turn point to the exception that caused <em>it</em>, and so on.</p>

